### PR TITLE
Fixed: Expanding json schema causes stack level too deep error

### LINKED ISSUE #50: Expanding json schema causes stack level too deep error
Here is the brief POC code.

```
json = <<_JSON_
{
    "id": "http://example.com/poc.json",
    "$schema": "http://json-schema.org/draft-04/hyper-schema#",
    "definitions": {
        "self":    { "$ref": "#" },
        "synonym": { "$ref": "#/definitions/self" }
    },
    "properties": {
        "id":   { "type": "string" }
    }
}
_JSON_

schema = JsonSchema::Parser.new.parse!(JSON.parse(json))
schema.expand_references #=> SystemStackError
```

I found `#/definitions/synonym` is treated as original, and `schema_children` of it includes `#/definitions/synonym`.
It means `schema.parent` returns schema itself, and it cause infinite loop and stack level too deep error when `schema.parent` is called against `#/definitions/synonym`.
(See: https://github.com/brandur/json_schema/blob/master/lib/json_schema/reference_expander.rb#L100-L104)

I also found `#` were not treated as original. This is also wired.

I tried to fix "original problem", but I can't....

### PR DESCRIPTION
**Pull request by okitan**
fixed #50

### CODE REVIEW

File: test/json_schema/reference_expander_test.rb [hunk 1]
```diff
@@ -173,6 +174,12 @@
     # but diving deeper results in the same pointer again
     schema = schema.properties["app"]
     assert_equal "#/properties/app", schema.pointer
+
+    schema = @schema.properties["app1"]
+    assert_equal "#/properties/app1", schema.pointer
+
+    schema = schema.properties["app1"]
+    assert_equal "#/properties/app1", schema.pointer
```
Comment (brandur) [inline test/json_schema/reference_expander_test.rb]: Aren't lines 181-182 identical to lines 178-179?

**Comment by brandur** [comment]
Nice! Thanks for taking a look at this.

File: test/json_schema/reference_expander_test.rb (hunk 1 above)
Comment (okitan) [inline test/json_schema/reference_expander_test.rb]: I didn't understand what this test did, and I copied https://github.com/okitan/json_schema/blob/fix_50/test/json_schema/reference_expander_test.rb#L170-L176 .

Do you think it is better to remove?

Comment (brandur) [inline test/json_schema/reference_expander_test.rb]: Sorry, my mistake. I thought both sets were using an instance variable `@schema`, but now see that's not the case.

**Comment by brandur** [comment]
Thanks again @okitan! Released as 0.12.1.

**Pull request merged**